examples/3d/generate_custom_mesh.rs

```diff
@@ -0,0 +1,261 @@
+        ..default()
+    });
+
+    // Light up the scene.
+    commands.spawn(PointLightBundle {
+        point_light: PointLight {
+            intensity: 1000.0,
+            range: 100.0,
+            ..default()
+        },
+        transform: camera_and_light_transform,
+        ..default()
+    });
+
+    // Text to describe the controls.
+    commands.spawn(
+        TextBundle::from_section(
+            "Controls:\nSpace: Change UVs\nX/Y/Z: Rotate\nR: Reset orientation",
+            TextStyle {
+                font_size: 20.0,
+                ..default()
+            },
+        )
+        .with_style(Style {
+            position_type: PositionType::Absolute,
+            top: Val::Px(12.0),
+            left: Val::Px(12.0),
+            ..default()
+        }),
+    );
+}
+
+// System to recieve input from the user, check out examples/input/ for more examples about user
+// input.
+fn input_handler(
+    keyboard_input: Res<Input<KeyCode>>,
+    mesh_query: Query<&Handle<Mesh>, With<CustomUV>>,
+    mut meshes: ResMut<Assets<Mesh>>,
+    mut query: Query<&mut Transform, With<CustomUV>>,
+    time: Res<Time>,
+) {
+    if keyboard_input.just_pressed(KeyCode::Space) {
+        let mesh_handle = mesh_query.get_single().expect("Query not successful");
+        let mesh = meshes.get_mut(mesh_handle).unwrap();
+        toggle_texture(mesh);
+    }
+    if keyboard_input.pressed(KeyCode::X) {
+        for mut transform in &mut query {
+            transform.rotate_x(time.delta_seconds() / 1.2);
+        }
+    }
+    if keyboard_input.pressed(KeyCode::Y) {
+        for mut transform in &mut query {
+            transform.rotate_y(time.delta_seconds() / 1.2);
+        }
+    }
+    if keyboard_input.pressed(KeyCode::Z) {
+        for mut transform in &mut query {
+            transform.rotate_z(time.delta_seconds() / 1.2);
+        }
+    }
+    if keyboard_input.pressed(KeyCode::R) {
+        for mut transform in &mut query {
+            transform.look_to(Vec3::NEG_Z, Vec3::Y);
+        }
+    }
+}
+
+fn create_cube_mesh() -> Mesh {
+    let mut cube_mesh = Mesh::new(PrimitiveTopology::TriangleList);
+
+    #[rustfmt::skip]
+    cube_mesh.insert_attribute(
+        Mesh::ATTRIBUTE_POSITION,
+        // Each array is an [x, y, z] coordinate in local space.
+        // Meshes always rotate around [0, 0, 0]
+        // By centering our mesh around the origin, rotation preserves the center of mass.
+        vec![
+            // top (facing towards +y)
+            [-0.5, 0.5, -0.5], // vertex with index 0
+            [0.5, 0.5, -0.5], // vertex wtih index 1
+            [0.5, 0.5, 0.5], // etc. until 23
+            [-0.5, 0.5, 0.5],
+            // bottom   (-y)
+            [-0.5, -0.5, -0.5],
+            [0.5, -0.5, -0.5],
+            [0.5, -0.5, 0.5],
+            [-0.5, -0.5, 0.5],
+            // right    (+x)
+            [0.5, -0.5, -0.5],
+            [0.5, -0.5, 0.5],
+            [0.5, 0.5, 0.5],
```

Comment:
```suggestion
            [0.5, 0.5, 0.5], // This vertex is at the same position as vertex with index 2, but they'll have different UV and normal
```
To clarify? idk, fine not putting it